specs/experimental/gov-delegation.md

```diff
@@ -309,17 +305,20 @@ enum AllowanceType {
 | `Absolute`  | `0`    | The amount of votes delegated is fixed and denominated in the `GovernanceToken`'s decimals  |
 | `Relative`  | `1`    | The amount of votes delegated is relative and denominated in percentages.                   |
 
-## Backwards Compatibility
-
-The `GovernanceDelegation` contract ensures backwards compatibility by allowing the migration of delegation state from the
-`GovernanceToken`.
-
 ## Migration
 
 All write functions in the `GovernanceDelegation` MUST check if the users interacting with it have been migrated by
 checking the `migrated` mapping from its [storage](#storage). If a user has not been migrated, the `GovernanceDelegation`
 MUST copy the delegation and checkpoint data from the token contract to its own state. After copying the data, the
-`GovernanceDelegation` MUST update the `migrated` mapping to reflect that the address has been migrated.
+`GovernanceDelegation` MUST update the `migrated` mapping to reflect that the address has been migrated, and remove the
+state from the `GovernanceToken` contract via a function that can only be called by the `GovernanceDelegation` contract.
```

Comment:
Function can be callable by any address. Basically just have a function like:

```solidity
function migrate(address _addr) public {
    address tmp = delegates[_addr];
    if (tmp != address(0)) {
        _delegate(_addr, address(0);
        governanceDelegation.delegateFromToken(address, tmp);
    }
}
```

Reply:
Must call this function whenever the user tries to create a delegation.

Reply:
Please also include an invariant in this spec that says that the user MUST NOT have an active delegation in both contracts at the same time. It's one or the other. We can say that a delegation to `address(0)` is an inactive delegation.

Reply:
agreed. added in https://github.com/ethereum-optimism/specs/pull/342/commits/fd3c52df61b48f751f3ba72636162cf27f501ec8